tests/integration/test_api_gateway.py

```diff
@@ -798,6 +800,87 @@ def test_import_rest_api(self):
         # clean up
         client.delete_rest_api(restApiId=rest_api_id)
 
+    def test_step_function_integrations(self):
+        client = aws_stack.connect_to_service('apigateway')
+        sfn_client = aws_stack.connect_to_service('stepfunctions')
+        lambda_client = aws_stack.connect_to_service('lambda')
+
+        state_machine_name = 'test'
+        state_machine_def = {
+            'Comment': 'Hello World example',
+            'StartAt': 'step1',
+            'States': {
+                'step1': {
+                    'Type': 'Task',
+                    'Resource': '__tbd__',
+                    'End': True
+                },
+            }
+        }
+
+        fn_name = 'test-stepfunctions-apigw'
+
+        # create state machine
+        testutil.create_lambda_function(
+            handler_file=TEST_LAMBDA_ECHO_FILE,
+            func_name=fn_name,
+            runtime=LAMBDA_RUNTIME_PYTHON36
+        )
+
+        role_arn = aws_stack.role_arn('sfn_role')
+
+        definition = clone(state_machine_def)
+        lambda_arn_1 = aws_stack.lambda_function_arn(fn_name)
+        definition['States']['step1']['Resource'] = lambda_arn_1
+        definition = json.dumps(definition)
+        sm_arn = 'arn:aws:states:us-east-1:%s:stateMachine:%s' \
+            % (TEST_AWS_ACCOUNT_ID, state_machine_name)
+
+        sfn_client.create_state_machine(
+            name=state_machine_name, definition=definition, roleArn=role_arn)
+
+        rest_api = client.create_rest_api(
+            name='test',
+            description='test'
+        )
+
+        resources = client.get_resources(
+            restApiId=rest_api['id']
+        )
+
+        client.put_method(
+            restApiId=rest_api['id'],
+            resourceId=resources['items'][0]['id'],
+            httpMethod='POST',
+            authorizationType='NONE'
+        )
+
+        client.put_integration(
+            restApiId=rest_api['id'],
+            resourceId=resources['items'][0]['id'],
+            httpMethod='POST',
+            integrationHttpMethod='POST',
+            type='AWS',
+            uri='arn:aws:apigateway:us-east-1:states:action/StartExecution',
+            requestTemplates={
+                'application/json': """
+                #set($data = $util.escapeJavaScript($input.json('$')))
+                {"input": "$data","stateMachineArn": "%s"}
+                """ % sm_arn
+            },
+        )
+
+        client.create_deployment(restApiId=rest_api['id'], stageName='dev')
+        url = gateway_request_url(api_id=rest_api['id'], stage_name='dev', path='/')
+        test_data = {'test': 'test-value'}
+        resp = requests.post(url, data=json.dumps(test_data))
+        self.assertEqual(resp.status_code, 200)
```

Comment:
Can we also add an assertion about the content (`resp.content`) that is being returned from the invocation here..?

Reply:
Added: content assertion